### ISSUE TITLE
Benchmark proteus on AI/ML codes 1

### ISSUE THREAD

**Issue by ggeorgakoudis** [Olympus-HPC/proteus#19]
- [x] Select ML benchmarks to run:
    - [x] Test cases: Non-contiguous buffers, sizes that are so large they [leave the bounds](https://github.com/pytorch/pytorch/blob/c81d4fd0a8ca826c165fe15f83398dbc4e20b523/aten/src/ATen/native/cuda/PersistentSoftmax.cuh#L305) of the fast implementations
    - [x] GPT (LLM / Transformer)
    - [x] CosmoFlow (3D convolutional network)
- [x] Apply proteus annotations to select operators in LBANN and PyTorch
    - [x] LBANN [layer normalization](https://github.com/tbennun/lbann/blob/develop/src/layers/regularizers/layer_norm.cu#L579)
    - [x] LBANN [softmax](https://github.com/LLNL/lbann/blob/develop/src/layers/activations/softmax.cu) -- **Crashes**
    - [x] ~~Other bottleneck LBANN operators? Update after profiling~~
(https://github.com/pytorch/pytorch/blob/c81d4fd0a8ca826c165fe15f83398dbc4e20b523/aten/src/ATen/native/cuda/CUDALoops.cuh#L183) kernels
    - [x] ~~Other bottleneck PyTorch operators? Update after profiling~~
- [x] Test run with proteus

* Evaluate with/without specialization
  * Performance
  * JIT latency / overhead
  * Memory overhead for persistent cache

**Comment by ggeorgakoudis** [comment]
@tbennun This is going well but it seems as if we're behind our deadline. Can we document what we have working now and add our forks are Olympus-HPC to open an issue associated with a different milestone. Open to suggestions, let's sync offline.

**Comment by tbennun** [comment]
We are probably on schedule but I seem to have misunderstood the issue as a global tracker, and put too many tasks on one issue. Anything that is marked "after profiling", for example, is beyond the first iteration.

We should discuss again how/where I push the benchmarks.
